app/org/sagebionetworks/bridge/services/UploadCertificateServiceImpl.java

```diff
@@ -52,19 +56,29 @@ public void createCmsKeyPair(final String studyIdentifier) {
                 .city(CONFIG.getProperty("upload.cms.certificate.city"))
                 .organization(CONFIG.getProperty("upload.cms.certificate.organization"))
                 .team(CONFIG.getProperty("upload.cms.certificate.team"))
-                .email(CONFIG.getProperty("upload.cms.certificate.email"))
-                .fqdn(CONFIG.getWebservicesURL())
-                .build();
+                .email(CONFIG.getProperty("upload.cms.certificate.email")).fqdn(CONFIG.getWebservicesURL()).build();
         final X509Certificate cert = certificateFactory.newCertificate(keyPair, certInfo);
-        final String name = studyIdentifier + ".pem";
+        final String name = studyIdentifier.getIdentifier() + ".pem";
         try {
-            s3Put(PRIVATE_KEY_BUCKET, name,  PemUtils.toPem(keyPair.getPrivate()));
+            s3Put(PRIVATE_KEY_BUCKET, name, PemUtils.toPem(keyPair.getPrivate()));
             s3Put(CERT_BUCKET, name, PemUtils.toPem(cert));
         } catch (CertificateEncodingException e) {
             throw new RuntimeException(e);
         }
     }
 
+    @Override
+    public String getPublicKeyAsPem(StudyIdentifier studyIdentifier) {
+        try {
+            GetObjectRequest request = new GetObjectRequest(CERT_BUCKET, studyIdentifier.getIdentifier() + ".pem");
+            S3Object object = s3CmsClient.getObject(request);
+
+            return IOUtils.toString(object.getObjectContent(), "UTF-8");
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
I don't like re-throwing as a RuntimeException. In testing, it's difficult to distinguish between this and a NullPointerException, for example. You should wrap it in a BridgeServiceException or equivalent.